### PR TITLE
Support Xcode 12 in CI

### PR DESCRIPTION
**Pull request by mokagio**
This PR adds support for Xcode 12 in CI. Notably, we had to bump the minimum deployment targets are recommended by Xcode 12, which results in a swarm of deprecation warnings, which I haven't had the time to look into, yet.

### CODE REVIEW

File: Simperium.xcodeproj/xcshareddata/xcschemes/Simperium iOS.xcscheme [hunk 1]
```diff
@@ -27,18 +27,9 @@
...
-            BlueprintIdentifier = "B5CAA4B31CAAB369006FE048"
-            BuildableName = "Simperium.framework"
-            BlueprintName = "Simperium_iOS"
-            ReferencedContainer = "container:Simperium.xcodeproj">
-         </BuildableReference>
-      </MacroExpansion>
       <Testables>
          <TestableReference
-            skipped = "NO">
+            skipped = "YES">
```
Comment (mokagio) [inline Simperium.xcodeproj/xcshareddata/xcschemes/Simperium iOS.xcscheme:32]: Temporarily disabled because they timeout on CI.

**Comment by jleandroperez** [comment]
### Quick update!
Depends on (quite a few!!) PR(s):

#597: Updates Core Data Threading model
#598: Fixes DiffMatchPatch Warnings
#599: Fixess RSSwizzle Warnings
#600: Fixes Xcode 12 Warnings
#601: Drops NSURLConnection

Will get back to you as soon as we're good to go!! Thank you Gio!

**Comment by mokagio** [comment]
Wow! There's a lot of moving pieces here. Thanks for tracking all of them down. I'll keep an eye on this 👍

File: Simperium.xcodeproj/xcshareddata/xcschemes/Simperium iOS.xcscheme (hunk 1 above)
Comment (jleandroperez) [reply Simperium.xcodeproj/xcshareddata/xcschemes/Simperium iOS.xcscheme:32]: I'll do my best to get to this during Hack Week!! Thanks sir!

**Comment by jleandroperez** [comment]
@mokagio quick update!

- Dependencies merged!
- There are a couple breaking Unit Tests (randomly)

I'm trying to get to the very bottom... since one of the latest PRs included a Core Data critical change, and I'd very much love to rule out any kind of threading issues.

Will keep you posted, apologies about crashing your PR!!

**Review by jleandroperez — APPROVED** [review]
Allright! Unit tests stabilized. Thank you for doing this @mokagio !!

**Comment by mokagio** [comment]
> Thank you for doing this @mokagio !!

I feel like I've done very little in this huge quest to support Xcode 12. Thanks ⭐ **YOU** ⭐

**Comment by jleandroperez** [comment]
**Actually** thanks to your CI integration... I've spotted a sync issue we've been hunting for months (haven't filed the actual bug yet).

**Thank you!!!**

**Pull request merged**